.github/workflows/main.yml

```diff
@@ -0,0 +1,94 @@
+        browser: [chromium, firefox]
+        os: [ubuntu-18.04]
+        nodeversion: ['16']
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.nodeversion }}
+    - name: Setup custom host for sdkapp.example.com
+      run: echo "127.0.0.1 sdkapp.example.com sdks.mytestrun.com sdkapp.petrov.ca auth.example.com openig.example.com api.example.com user.example.com rest.example.com bank.example.com openam.example.com spa.example.com sdk.mytestrun.com sdks.mytestrun.com forgerock-sdk.local" | sudo tee -a /etc/hosts
+    - name: Install NPM
+      run: npm i -g npm@8
+    - name: Install Dependencies
+      run: npm ci
+      env:
+        DEBUG: pw:install
+        PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1
+    - name: Build SDK
+      run: npm run build
+    - name: Install Playwright
+      run: npx playwright install --with-deps
+    - name: Run e2e
+      run: npm run test:e2e --project=${{ matrix.browser }}
+      env:
+        NODE_TLS_REJECT_UNAUTHORIZED: 0  
+  osx:
+      name: ${{ matrix.os }} (${{ matrix.browser }})
+      strategy:
+        fail-fast: false
+        matrix:
+          browser: [chromium, firefox, webkit]
+          os: [macos-11.0]
+          nodeversion: ['16']
+      runs-on: ${{ matrix.os }}
+      steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.nodeversion }}
+      - name: Setup custom host for sdkapp.example.com
+        run: echo "127.0.0.1 sdkapp.example.com sdks.mytestrun.com sdkapp.petrov.ca auth.example.com openig.example.com api.example.com user.example.com rest.example.com bank.example.com openam.example.com spa.example.com sdk.mytestrun.com sdks.mytestrun.com forgerock-sdk.local" | sudo tee -a /etc/hosts      
+      - name: Install NPM
+        run: npm i -g npm@8
+      - name: Install Dependencies
+        run: npm ci
+        env:
+          DEBUG: pw:install
+          # PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1
+      # - run: npx playwright install --with-deps ${{ matrix.browser }} chromium
+      - run: npm run build
+      - run: npm run test:e2e
+  windows:
+    name: windows (${{ matrix.browser }})
+    strategy:
+      fail-fast: false
+      matrix:
+        browser: [chromiumm, msedge, firefox] # firefox, webkit, msedge]
+        nodeversion: ['16']
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.nodeversion }}
+    - name: Setup custom host for sdkapp.example.com
+      run: echo "`n127.0.0.1  sdkapp.example.com sdks.mytestrun.com sdkapp.petrov.ca auth.example.com openig.example.com api.example.com user.example.com rest.example.com bank.example.com openam.example.com spa.example.com sdk.mytestrun.com sdks.mytestrun.com forgerock-sdk.local" >> C:\Windows\System32\drivers\etc\hosts   #add to new line
```

Comment:
windows way to edit etc/hosts file. its ugly but it gets the job done.

Reply:
Nice job!

With all of these, let's reduce the domain aliases to just the required ones. There's a lot of legacy domains in here that I'd hate to carry it into our new process.